runtime/stdlib/cadence_v0.42_to_v1_contract_upgrade_validator.go

```diff
@@ -499,6 +533,92 @@ func (validator *CadenceV042ToV1ContractUpdateValidator) checkDeclarationKindCha
 	return false
 }
 
+func (validator *CadenceV042ToV1ContractUpdateValidator) checkConformanceV1(
+	oldDecl *ast.CompositeDeclaration,
+	newDecl *ast.CompositeDeclaration,
+) {
+
+	// Here it is assumed enums will always have one and only one conformance.
+	// This is enforced by the checker.
+	// Therefore, below check for multiple conformances is only applicable
+	// for non-enum type composite declarations. i.e: structs, resources, etc.
+
+	oldConformances := oldDecl.Conformances
+
+	location := validator.underlyingUpdateValidator.location
+
+	elaboration := validator.newElaborations[location]
+	newDeclType := elaboration.CompositeDeclarationType(newDecl)
+
+	// A conformance may not be explicitly defined in the current declaration,
+	// but they could be available via inheritance.
+	newConformances := newDeclType.EffectiveInterfaceConformances()
+
+	// All the existing conformances must have a match. Order is not important.
+	// Having extra new conformance is OK. See: https://github.com/onflow/cadence/issues/1394
+
+	// Note: Removing a conformance is NOT OK. That could lead to type-safety issues.
+	// e.g:
+	//  - Someone stores an array of type `[{I}]` with `T:I` objects inside.
+	//  - Later T’s conformance to `I` is removed.
+	//  - Now `[{I}]` contains objects if `T` that does not conform to `I`.
+
+	for _, oldConformance := range oldConformances {
+		found := false
+		for index, newConformance := range newConformances {
+			newConformanceNominalType := semaConformanceToASTNominalType(newConformance)
+
+			err := oldConformance.CheckEqual(newConformanceNominalType, validator)
+
+			var customRuleChecked, customRuleValid bool
+			if err != nil {
+				customRuleChecked, customRuleValid =
+					validator.checkUserDefinedTypeCustomRules(oldConformance, newConformanceNominalType)
+			}
```

Comment:
If the usual type equality fails, also check in the special rules to see if this type change is a valid one.
The value migrations (static type migration etc.) already use these rules to migrate value. With this change, contract update validator becomes consistent with the value migrations.

(Rest of the logic is same as the old contract update validator)